conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -1,195 +1,261 @@
-        self.assertEqual(cppstd_default("gcc", "7.3"), "gnu14")
-        self.assertEqual(cppstd_default("gcc", "8.1"), "gnu14")
-
-    def test_clang_cppstd_flags(self):
-        self.assertEqual(cppstd_flag("clang", "2.0", "98"), None)
-        self.assertEqual(cppstd_flag("clang", "2.0", "gnu98"), None)
-        self.assertEqual(cppstd_flag("clang", "2.0", "11"), None)
-        self.assertEqual(cppstd_flag("clang", "2.0", "14"), None)
-
-        self.assertEqual(cppstd_flag("clang", "2.1", "98"), "-std=c++98")
-        self.assertEqual(cppstd_flag("clang", "2.1", "gnu98"), "-std=gnu++98")
-        self.assertEqual(cppstd_flag("clang", "2.1", "11"), "-std=c++0x")
-        self.assertEqual(cppstd_flag("clang", "2.1", "14"), None)
-
-        self.assertEqual(cppstd_flag("clang", "3.0", "11"), '-std=c++0x')
-        self.assertEqual(cppstd_flag("clang", "3.0", "14"), None)
-
-        self.assertEqual(cppstd_flag("clang", "3.1", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "3.1", "14"), None)
-
-        self.assertEqual(cppstd_flag("clang", "3.4", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "3.4", "14"), '-std=c++1y')
-        self.assertEqual(cppstd_flag("clang", "3.4", "17"), None)
-
-        self.assertEqual(cppstd_flag("clang", "3.5", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "3.5", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("clang", "3.5", "17"), '-std=c++1z')
-
-        self.assertEqual(cppstd_flag("clang", "5", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "5", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("clang", "5", "gnu14"), '-std=gnu++14')
-        self.assertEqual(cppstd_flag("clang", "5", "17"), '-std=c++17')
-
-        self.assertEqual(cppstd_flag("clang", "5.1", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "5.1", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("clang", "5.1", "17"), '-std=c++17')
-
-        self.assertEqual(cppstd_flag("clang", "6", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "6", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("clang", "6", "17"), '-std=c++17')
-        self.assertEqual(cppstd_flag("clang", "6", "20"), '-std=c++2a')
-
-        self.assertEqual(cppstd_flag("clang", "7", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "7", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("clang", "7", "17"), '-std=c++17')
-        self.assertEqual(cppstd_flag("clang", "7", "20"), '-std=c++2a')
-
-        self.assertEqual(cppstd_flag("clang", "8", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("clang", "8", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("clang", "8", "17"), '-std=c++17')
-        self.assertEqual(cppstd_flag("clang", "8", "20"), '-std=c++2a')
+        with self.assertRaises(ConanException):
+            make_cppstd_default("gcc", "4.0")
```

Comment:
```suggestion
```

Failing because of the helper function, not because of the underlying `cppstd_default` being tested here

Reply:
removed